src/lib/operators/table_scan/like_table_scan_impl.cpp

```diff
@@ -131,6 +90,149 @@ void LikeTableScanImpl::handle_column(const BaseDictionaryColumn& base_column,
   });
 }
 
+LikeTableScanImpl::PatternTokens LikeTableScanImpl::pattern_string_to_tokens(const std::string& pattern) {
+  PatternTokens tokens;
+
+  auto current_position = size_t{0};
+  while (current_position < pattern.size()) {
+    if (pattern[current_position] == '_') {
+      tokens.emplace_back(Wildcard::SingleChar);
+      ++current_position;
+    } else if (pattern[current_position] == '%') {
+      tokens.emplace_back(Wildcard::AnyChars);
+      ++current_position;
+    } else {
+      const auto next_wildcard_position = pattern.find_first_of("_%", current_position);
+      const auto token_length =
+          next_wildcard_position == std::string::npos ? std::string::npos : next_wildcard_position - current_position;
+      tokens.emplace_back(pattern.substr(current_position, token_length));
+      current_position = next_wildcard_position;
+    }
+  }
+
+  return tokens;
+}
+
+LikeTableScanImpl::AllPatternVariant LikeTableScanImpl::pattern_string_to_pattern_variant(const std::string& pattern) {
+  const auto tokens = pattern_string_to_tokens(pattern);
+
+  if (tokens.size() == 2 && tokens[0].type() == typeid(std::string) && tokens[1] == PatternToken{Wildcard::AnyChars}) {
+    return StartsWithPattern{boost::get<std::string>(tokens[0])};
+  } else if (tokens.size() == 2 && tokens[0] == PatternToken{Wildcard::AnyChars} &&  // NOLINT
+             tokens[1].type() == typeid(std::string)) {
+    return EndsWithPattern{boost::get<std::string>(tokens[1])};
+  } else if (tokens.size() == 3 && tokens[0] == PatternToken{Wildcard::AnyChars} &&  // NOLINT
+             tokens[1].type() == typeid(std::string) && tokens[2] == PatternToken{Wildcard::AnyChars}) {
+    return ContainsPattern{boost::get<std::string>(tokens[1])};
+  } else {
+    // Pick ContainsMultiple or Regex
+    auto impl_is_contains_multiple = true;      // Set to false if tokens don't match %{, str, %} pattern
+    auto strings = std::vector<std::string>{};  // arguments used for ContainsMultiple, if it gets used
+    auto expect_any_chars = true;               // If false, expect a string
```

Comment:
The MultipleContainsPattern part is a bit hard for me to understand. The variable names don't help me too much (expect any char... where? Also, char or chars?) - could you give a high-level explanation of this?